homeassistant/components/roomba/__init__.py

```diff
@@ -167,9 +174,16 @@ async def async_unload_entry(hass, config_entry):
             ]
         )
     )
+
+    domain_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    # unsubscribe listerner
+    domain_data[LISTENER]()
+
+    # disconnect cleanly
+    await async_disconnect_or_timeout(hass, roomba=domain_data[ROOMBA_SESSION])
```

Comment:
We're not catching `CannotDisconnect` here. We need to catch any exceptions we raise here.

Reply:
We also need to catch that new error in the config flow. And that needs to be tested.

Reply:
> We're not catching `CannotDisconnect` here. We need to catch any exceptions we raise here.
I understand but I must disconnect the session from roomba because otherwise it remains in memory and it is impossible to initiate a new one.
In this case, where should I put my disconnect call ? at the start of the function ?

Reply:
Why not catch the error here and return False in that  case?

Reply:
I am stupid, you are right it is enough that I return false. I just pushed the change